src/NavBar.ts

```diff
@@ -112,60 +88,78 @@ export default class NavBar {
-        const bbox = this.getBBoxFromUrlPoints(parsedPoints.map(p => p.coordinate))
-        if (bbox) Dispatcher.dispatch(new SetInitialBBox(bbox))
+        if (parsedProfileName)
+            // this won't trigger a route request because we just cleared the points
+            Dispatcher.dispatch(new SetVehicleProfile({ name: parsedProfileName }))
+        const parsedPoints = NavBar.parsePoints(url)
 
-        if (parsedPoints.length > 0) Dispatcher.dispatch(new SetRoutingParametersAtOnce(parsedPoints, profile))
+        // support legacy URLs without coordinates (not initialized) and only text, see #199
+        if (parsedPoints.some(p => !p.isInitialized && p.queryText.length > 0)) {
+            const promises = parsedPoints.map(p => {
+                if (p.isInitialized) return Promise.resolve(p)
+                return (
+                    getApi()
+                        .geocode(p.queryText, 'nominatim')
+                        .then(res => {
+                            if (res.hits.length == 0) return p
+                            return {
+                                ...p,
+                                queryText: res.hits[0].name,
+                                coordinate: { lat: res.hits[0].point.lat, lng: res.hits[0].point.lng },
+                                isInitialized: true,
+                            }
+                        })
+                        // if the geocoding request fails we just keep the point as it is, just as if no results were found
+                        .catch(() => Promise.resolve(p))
+                )
+            })
+            const points = await Promise.all(promises)
+            NavBar.dispatchQueryPoints(points)
+        } else {
+            NavBar.dispatchQueryPoints(parsedPoints)
+        }
 
-        // add map style
-        const parsedStyleOption = this.parseLayer(url)
-        Dispatcher.dispatch(new SelectMapStyle(parsedStyleOption))
+        const parsedLayer = NavBar.parseLayer(url)
+        if (parsedLayer) Dispatcher.dispatch(new SelectMapLayer(parsedLayer))
 
-        this.isIgnoreQueryStoreUpdates = false
+        this.ignoreStateUpdates = false
     }
 
-    private fillPoints(parsedPoints: QueryPoint[]) {
-        const result: QueryPoint[] = this.queryStore.state.queryPoints
-
-        // assuming that at least two points should be present add un-initialized points if necessary
-        for (let i = 0; i < result.length && i < parsedPoints.length; i++) {
-            result[i] = parsedPoints[i]
-        }
-
-        return result
+    private static dispatchQueryPoints(points: QueryPoint[]) {
+        // estimate map bounds from url points if there are any. this way we prevent loading tiles for the world view
+        // only to zoom to the route shortly after
+        const bbox = NavBar.getBBoxFromUrlPoints(points.filter(p => p.isInitialized).map(p => p.coordinate))
+        if (bbox) Dispatcher.dispatch(new SetInitialBBox(bbox))
```

Comment:
We can do this for urls with `point=queryString' as well now, even though it doesn't help as much, because the (world) map already starts to load while the slow geocoding requests are still ongoing.